staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1040,12 +1040,22 @@ func (c *cacheWatcher) process(initEvents []*watchCacheEvent, resourceVersion ui
 		klog.V(2).Infof("processing %d initEvents of %s took %v", len(initEvents), objType, processingTime)
 	}
 
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
```

Comment:
Is it correct - if we pass timeout 0, we probably don't want to close it immediately - I think timeout 0 should be treated differently.

Reply:
I'd like to treat 0 as never.